extras/create_snapshots_reports_scorecard.py

```diff
@@ -337,46 +387,138 @@ def generate_snapshots_from_list(db, cyhy_db_section):
+            # Grouping nodes are treated as third-party orgs when creating
+            # snapshots since both require the "--use-only-existing-snapshots"
+            # flag to be set.
+            snapshot_rc = generate_snapshot(
+                db, cyhy_db_section, grouping_node_id, third_party=True
+            )
+
+            if snapshot_rc != 0:
+                logging.error(
+                    "Grouping node %s snapshot creation failed!", grouping_node_id
+                )
+                logging.error(
+                    "Third-party snapshots (dependent on %s) cannot be created for: %s",
+                    grouping_node_id,
+                    tp_dependence_map[grouping_node_id],
+                )
+                # Add dependent third-party snapshot org IDs to failed list and
+                # remove them from list of tp_reports_to_generate so that we
+                # don't attempt to create reports for them later.
+                global failed_tp_snapshots
+                for tp_org_id in tp_dependence_map[grouping_node_id]:
+                    if tp_org_id not in failed_tp_snapshots:
+                        failed_tp_snapshots.append(tp_org_id)
+                    if tp_org_id in tp_reports_to_generate:
+                        tp_reports_to_generate.remove(tp_org_id)
+    else:
+        logging.info("No grouping node snapshots needed for third-party reports")
+    
+    time_to_generate_grouping_node_snapshots = time.time() - start_time
+    logging.info(
+        "Time to complete grouping node snapshots: %.2f minutes",
+        time_to_generate_grouping_node_snapshots / 60,
+    )
+    return sorted(list(tp_reports_to_generate)), time_to_generate_grouping_node_snapshots
+
+
+def manage_snapshot_threads(db, cyhy_db_section, third_party):
     """Spawn threads to generate snapshots
     
-    Build the lists of reports and snapshots to be generated, then spawn the
-    threads that generate the snapshots."""
+    Build the list of snapshots to be generated, then spawn the threads that
+    generate the snapshots."""
     start_time = time.time()
 
-    logging.info("Building list of reports to generate...")
-    reports_to_generate = create_list_of_reports_to_generate(db)
+    # This variable is only used when generating third-party snapshots
+    time_to_generate_grouping_node_snapshots = 0
 
-    logging.info("Building list of snapshots to generate...")
+    global reports_to_generate
     global snapshots_to_generate
-    # No thread locking is needed here for snapshots_to_generate because we are
-    # still single-threaded at this point
-    snapshots_to_generate = create_list_of_snapshots_to_generate(
-        db, reports_to_generate
-    )
+    # No thread locking is needed here for reports_to_generate or
+    # snapshots_to_generate because we are still single-threaded at this point
+
+    if third_party:
+        # Some extra preparation must be done before generating third-party snapshots
```

Comment:
Might be nice to have some logging here to indicate which path we went down since the `else` logs.

Reply:
I didn't log anything there because it starts logging inside of `prepare_for_third_party_snapshots()`, but I can add something here just in case.

Reply:
mm yeah I neglected to account for logging in `prepare_for_third_party_snapshots()`. That should be fine.